src/GameBoard.hs

```diff
@@ -23,6 +23,7 @@ module GameBoard
 
 -- I want to use my own Vector.
 import Graphics.Gloss  hiding (Vector)
+import Data.Bool
```

Comment:
Do you actually use `Data.Bool` for anything ? This import doesn't seem justified by any of your changes (the `Bool` datatype is already in `Prelude`).